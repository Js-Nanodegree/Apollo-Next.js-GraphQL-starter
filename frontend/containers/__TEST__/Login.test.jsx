import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import LoginContainer from '../Login';

describe('Login container', () => {
  it('should match snap shot', () => {
    const handleLogin = jest.fn();
    const handleChange = jest.fn();

    const tree = renderer
      .create(
        <LoginContainer
          handleLogin={handleLogin}
          loading={false}
          email=''
          password=''
          handleChange={handleChange}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should disable the submit button if !email || !password', () => {
    const handleLogin = jest.fn();
    const handleChange = jest.fn();
    const render = shallow(
      <LoginContainer
        handleLogin={handleLogin}
        loading={false}
        email=''
        password=''
        handleChange={handleChange}
      />
    );

    const submitButton = render.find('[type="submit"]');

    expect(submitButton.prop('disabled')).toBe(true);
  });

  it('Should enable the submit button if email && password', () => {
    const handleLogin = jest.fn();
    const handleChange = jest.fn();
    const render = shallow(
      <LoginContainer
        handleLogin={handleLogin}
        loading={false}
        email='test@example.com'
        password='passeord'
        handleChange={handleChange}
      />
    );

    const submitButton = render.find('[type="submit"]');

    expect(submitButton.prop('disabled')).toBe(false);
  });

  it('Should call the handle login function', () => {
    const handleLogin = jest.fn(() => true);
    const handleChange = jest.fn();
    const render = shallow(
      <LoginContainer
        handleLogin={handleLogin}
        loading={false}
        email='test@example.com'
        password='passeord'
        handleChange={handleChange}
      />
    );

    const submitButton = render.find('[type="submit"]');

    submitButton.simulate('click');

    expect(handleLogin).toBeCalled();
  });
});
