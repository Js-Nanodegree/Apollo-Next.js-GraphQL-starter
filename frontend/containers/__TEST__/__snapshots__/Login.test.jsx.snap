// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login container should match snap shot 1`] = `
<div
  className="qepopj-0 gBRnsH"
>
  <div
    className="content-middle"
  >
    <form
      className="cw1g19-0 cGYWuk"
      onSubmit={[Function]}
    >
      <h1>
        Login
      </h1>
      <div
        className="MuiFormControl-root-1 MuiFormControl-fullWidth-4"
      >
        <label
          className="MuiFormLabel-root-16 MuiFormLabel-required-21 MuiInputLabel-required-9 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
          data-shrink={false}
        >
          Email
          <span
            className="MuiFormLabel-asterisk-22"
          >
             *
          </span>
        </label>
        <div
          className="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-fullWidth-45 MuiInput-fullWidth-30 MuiInputBase-formControl-37 MuiInput-formControl-24"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="email"
            className="MuiInputBase-input-46 MuiInput-input-31 MuiInputBase-inputType-49 MuiInput-inputType-34"
            disabled={false}
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="email"
            value=""
          />
        </div>
      </div>
      <br />
      <br />
      <div
        className="MuiFormControl-root-1 MuiFormControl-fullWidth-4"
      >
        <label
          className="MuiFormLabel-root-16 MuiFormLabel-required-21 MuiInputLabel-required-9 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
          data-shrink={false}
        >
          password
          <span
            className="MuiFormLabel-asterisk-22"
          >
             *
          </span>
        </label>
        <div
          className="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-fullWidth-45 MuiInput-fullWidth-30 MuiInputBase-formControl-37 MuiInput-formControl-24"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="current-password"
            className="MuiInputBase-input-46 MuiInput-input-31 MuiInputBase-inputType-49 MuiInput-inputType-34"
            disabled={false}
            name="password"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="password"
            value=""
          />
        </div>
      </div>
      <br />
      <br />
      <button
        className="MuiButtonBase-root-79 MuiButtonBase-disabled-80 MuiButton-root-53 MuiButton-contained-64 MuiButton-containedPrimary-65 MuiButton-raised-67 MuiButton-raisedPrimary-68 MuiButton-disabled-73 MuiButton-fullWidth-78"
        disabled={true}
        id="submit"
        onBlur={[Function]}
        onClick={[MockFunction]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="submit"
      >
        <span
          className="MuiButton-label-54"
        >
          Login
        </span>
      </button>
      <br />
      <p>
        <a
          href="/register"
          onClick={[Function]}
        >
          Register
        </a>
      </p>
    </form>
  </div>
</div>
`;
